connectors/kafka/src/main/java/forklift/connectors/KafkaConnector.java

```diff
@@ -0,0 +1,167 @@
+
+    private KafkaController createController() {
+        Properties props = new Properties();
+        props.put("bootstrap.servers", kafkaHosts);
+        props.put("group.id", groupId);
+        props.put("enable.auto.commit", false);
+        props.put("consumer.timeout.ms", "-1");
+        props.put("key.deserializer", io.confluent.kafka.serializers.KafkaAvroDeserializer.class);
+        props.put("value.deserializer", io.confluent.kafka.serializers.KafkaAvroDeserializer.class);
+        props.put("schema.registry.url", schemaRegistries);
+        props.put("specific.avro.reader", false);
+        //props.put("auto.offset.reset", "earliest");
+        this.kafkaConsumer = new KafkaConsumer(props);
+        this.controller = new KafkaController(kafkaConsumer, messageStream);
+        return controller;
+    }
+
+    @Override
+    public void stop() throws ConnectorException {
+        try {
+            this.controller.stop(2000, TimeUnit.MILLISECONDS);
+        } catch (InterruptedException e) {
+            log.error("KafkaConnector interrupted while stopping");
+        }
+        this.kafkaProducer.close();
+    }
+
+    @Override
+    public Connection getConnection() throws ConnectorException {
+        throw new UnsupportedOperationException("Kafka does not support Connections, please work with Queues and Topics");
+    }
+
+    @Override
+    public ForkliftConsumerI getQueue(String name) throws ConnectorException {
+        return this.getTopic(name);
+    }
+
+    @Override
+    public ForkliftConsumerI getTopic(String name) throws ConnectorException {
+        synchronized (this) {
+            if (this.controller == null || !this.controller.isRunning()) {
+                this.controller = createController();
+                this.controller.start();
+            }
+        }
+        return new KafkaTopicConsumer(name, controller, messageStream);
+    }
+
+    @Override
+    public ForkliftProducerI getQueueProducer(String name) {
+        return this.getTopicProducer(name);
+    }
+
+    @Override
+    public ForkliftProducerI getTopicProducer(String name) {
+        return new KafkaForkliftProducer(name, this.kafkaProducer);
+    }
+
+    @Override
+    public ForkliftMessage jmsToForklift(Message m) {
+        try {
+            final ForkliftMessage msg = new ForkliftMessage(m);
+            if (m instanceof KafkaMessage) {
+                KafkaMessage kafkaMessage = (KafkaMessage)m;
+                msg.setProperties(kafkaMessage.getProperties());
+                ConsumerRecord<?, ?> record = kafkaMessage.getConsumerRecord();
+                if (record.value() instanceof GenericRecord) {
+                    GenericRecord genericRecord = (GenericRecord)record.value();
+                    Object value = genericRecord.get("forkliftMapMsg");
+                    value = value == null ? genericRecord.get("forkliftMsg") : value;
+                    value = value == null ? genericRecord.get("forkliftJsonMsg") : value;
```

Comment:
Shouldn't we deserialize these values from our message so that our message's value appears to be what was sent? I don't see anything deserializing the string that was constructed in the producer for maps, for example.

Also, with removing the JMS stuff, this is going to need to find a new home...